mmv1/products/billingbudget/terraform.yaml

```diff
@@ -92,7 +101,33 @@ overrides: !ruby/object:Overrides::ResourceOverrides
         custom_flatten: "templates/terraform/custom_flatten/billing_budget_budget_filter_labels.erb"
       budgetFilter.projects: !ruby/object:Overrides::Terraform::PropertyOverride
         is_set: true
-
+      budgetFilter.calendarPeriod: !ruby/object:Overrides::Terraform::PropertyOverride
+        diff_suppress_func: 'emptyOrDefaultStringSuppress("MONTH")'
```

Comment:
Figured out what needs to be done with this.
Let's remove the diff suppress and add
`        default_from_api: true`
while also adding a custom expander. An example of which is [here](https://github.com/GoogleCloudPlatform/magic-modules/blob/1f591d91b01408185c83b3b982a00ac154fcf58f/mmv1/products/bigquery/terraform.yaml#L247) with the associated file [here](https://github.com/GoogleCloudPlatform/magic-modules/blob/main/mmv1/templates/terraform/custom_expand/bigquery_table_ref.go.erb)

The expander logic will be:
```
	if _, ok := d.GetOkExists("budget_filter.0.custom_period.0.start_date"); ok {
		return nil, nil
	}
	return v, nil
```

This will make the field default to the API if not present in Terraform. The expander will specify that we should send nothing to the API if `custom_period` is set